src/reference/asciidoc/reactive-streams.adoc

```diff
@@ -160,12 +160,101 @@ When the target protocol for integration provides a Reactive Streams solution, i
+        this.customReactiveSource = customReactiveSource;
+    }
+
+    @Override
+    protected void doStart() {
+        Flux < Message << ? >> messageFlux =
+            this.myReactiveSource
+            .map(event - >
+                MessageBuilder
+                .withPayload(event.getBody())
+                .setHeader(MyReactiveHeaders.SOURCE_NAME, event.getSourceName())
+                .build());
+
+        subscribeToPublisher(messageFlux);
+    }
+
+}
+```
+
+A reactive outbound channel adapter implementation is about the initiation (or continuation) of a reactive stream to interaction with an external system according to the provided reactive API for the target protocol.
+An inbound payload could be a reactive type per se or as an event of the whole integration flow which is a part of the reactive stream on top.
+A returned reactive type can be subscribed immediately if we are in a one-way, fire-and-forget scenario, or it is propagated downstream (request-reply scenarios) for further integration flow or an explicit subscription in the target business logic, but still downstream preserving reactive streams semantics.
+
+An example for an outbound channel adapter:
+```java
+public class CustomReactiveMessageHandler extends AbstractReactiveMessageHandler {
+
+    private final CustomEntityOperations customEntityOperations;
+
+    private Expression queryTypeExpression = new ValueExpression<>(Type.INSERT);
+
+    public CustomReactiveMessageHandler(CustomEntityOperations customEntityOperations, Expression queryTypeExpression) {
+        Assert.notNull(customEntityOperations, "'customEntityOperations' must not be null");
+        this.customEntityOperations = customEntityOperations;
+        if (queryTypeExpression != null) {
+            this.queryTypeExpression = queryTypeExpression;
+        }
+    }
+
+    @Override
+    protected Mono<Void> handleMessageInternal(Message<?> message) {
+        return Mono.fromSupplier(() -> this.queryTypeExpression.getValue(this.evaluationContext, message, Type.class))
+                .flatMap(mode -> {
+                    switch (mode) {
+                        case INSERT:
+                            return handleInsert(message);
+                        case UPDATE:
+                            return handleUpdate(message);
+                        default:
+                            return Mono.error(new IllegalArgumentException());
+                    }
+                }).then();
+    }
+
+    private Mono<Void> handleInsert(Message<?> message) {
+        return this.customEntityOperations.insert(message.getPayload())
+                .then();
+    }
+
+    private Mono<Void> handleUpdate(Message<?> message) {
+        return this.r2dbcEntityOperations.update(message.getPayload())
+                .then();
+    }
+
+    public enum Type {
+        INSERT,
+        UPDATE,
+    }
+}
+```
+
+And now we will be able to use it:
```

Comment:
I think this flow sample belongs to the `producer` section above.
The `CustomReactiveMessageHandler` deserves its own independent usage sample.
Otherwise it is confusing a little bit.

Reply:
Can be resolved